packages/next/shared/lib/side-effect.tsx

```diff
@@ -14,20 +14,20 @@ type SideEffectProps = {
 }
 
 const isServer = typeof window === 'undefined'
-const useClientOnlyLayoutEffect = isServer ? () => {} : useLayoutEffect
-const useClientOnlyEffect = isServer ? () => {} : useEffect
+const useClientOnlyLayoutEffect = isServer ? () => { } : useLayoutEffect
+const useClientOnlyEffect = isServer ? () => { } : useEffect
 
-export default function SideEffect(props: SideEffectProps) {
+const SideEffect = memo((props: SideEffectProps) => {
   const { headManager, reduceComponentsToState } = props
 
-  function emitChange() {
+  const emitChange = useCallback(() => {
     if (headManager && headManager.mountedInstances) {
       const headElements = Children.toArray(
         Array.from(headManager.mountedInstances as Set<unknown>).filter(Boolean)
       ) as React.ReactElement[]
       headManager.updateHead(reduceComponentsToState(headElements, props))
     }
-  }
+  }, [props, headManager, reduceComponentsToState])
```

Comment:
I thought only adding `props` as dependencies is enough but currently i added others because i am not 100% sure about it, please any suggestion.

Reply:
IMHO, wrapping `emitChange` in useCallback is not required.

`<SideEffect />` component do not have `useState` or `useSyncExternalStore`. So the component will only update when its prop changes. And if prop changes, so does the `emitChanges`.

Reply:
> IMHO, wrapping `emitChange` in useCallback is not required.
> 
> `<SideEffect />` component do not have `useState` or `useSyncExternalStore`. So the component will only update when its prop changes. And if prop changes, so do the `emitChanges`.

Ok :)